test/integration/transactionsAPITests.coffee

```diff
@@ -196,7 +207,119 @@ describe "API Integration Tests", ->
                 (newTransaction != null).should.be.true
                 newTransaction.request.body.length.should.be.exactly utils.MAX_BODIES_SIZE
                 newTransaction.canRerun.should.be.false
-                done()    
+                done()
+                
+      it  "should add a transaction and truncate the routes request body", (done) ->
+        # Given
+        td = JSON.parse JSON.stringify transactionData
+        td.channelID = channel._id
+        clearTransactionBodies(td)
+        td.routes[0].request.body = largeBody
+        
+        # When
+        request("https://localhost:8080")
+          .post("/transactions")
+          .set("auth-username", testUtils.rootUser.email)
+          .set("auth-ts", authDetails.authTS)
+          .set("auth-salt", authDetails.authSalt)
+          .set("auth-token", authDetails.authToken)
+          .send(td)
+          .expect(201)
+          .end (err, res) ->
+            if err
+              done err
+            else
+              # Then
+              Transaction.findOne { clientID: "999999999999999999999999" }, (error, newTransaction) ->
+                should.not.exist (error)
+                (newTransaction != null).should.be.true
+                newTransaction.routes[0].request.body.length.should.be.exactly utils.MAX_BODIES_SIZE
+                newTransaction.canRerun.should.be.true
+                done()
+      
+      it  "should add a transaction and truncate the routes response body", (done) ->
+        # Given
+        td = JSON.parse JSON.stringify transactionData
+        td.channelID = channel._id
+        clearTransactionBodies(td)
+        td.routes[0].response.body = largeBody
+        
+        # When
+        request("https://localhost:8080")
+          .post("/transactions")
+          .set("auth-username", testUtils.rootUser.email)
+          .set("auth-ts", authDetails.authTS)
+          .set("auth-salt", authDetails.authSalt)
+          .set("auth-token", authDetails.authToken)
+          .send(td)
+          .expect(201)
+          .end (err, res) ->
+            if err
+              done err
+            else
+              # Then
+              Transaction.findOne { clientID: "999999999999999999999999" }, (error, newTransaction) ->
+                should.not.exist (error)
+                (newTransaction != null).should.be.true
+                newTransaction.routes[0].response.body.length.should.be.exactly utils.MAX_BODIES_SIZE
+                newTransaction.canRerun.should.be.true
+                done()
+                
```

Comment:
Maybe add a test to define what should happen when both request/response body has a large payload